pandas/tests/series/test_missing.py

```diff
@@ -1398,3 +1400,47 @@ def test_nonzero_warning(self):
         ser = pd.Series([1, 0, 3, 4])
         with tm.assert_produces_warning(FutureWarning):
             ser.nonzero()
+
+    @pytest.mark.parametrize(
+        "ind",
+        [
+            ['a', 'b', 'c', 'd'],
+            pd.period_range(start="2019-01-01", periods=4),
+            pd.interval_range(start=0, end=4),
+            pd.timedelta_range(start=1, periods=4),
+        ])
+    def test_interpolate_non_numerical_index(self, nontemporal_method, ind):
+        """
+        Tests for non numerical index types  - object, period, timedelta
+        Note that except time, index, nearest and values methods all others
+        are tested here.
+        """
+        # gh 21662
+        df = pd.DataFrame([0, 1, np.nan, 3], index=ind)
+
+        method, kwargs = nontemporal_method
+        if method == "pchip":
+            _skip_if_no_pchip()
+        elif is_timedelta64_dtype(ind):
+            if method in {"linear", "index", "values", "pchip"}:
```

Comment:
i find this a touch complicated. it might be worthile to make this into several tests (you would make a fixture for the input parameterization)